src/dash-table/derived/cell/contents.tsx

```diff
@@ -58,63 +60,69 @@ class Contents {
     get = memoizeOne((
         activeCell: ActiveCell,
         columns: VisibleColumns,
+        defaultLocale: LocaleFormat,
         data: Data,
         offset: IViewportOffset,
         editable: boolean,
         isFocused: boolean,
         dropdowns: (DropdownValues | undefined)[][]
-    ): JSX.Element[][] => mapData(
-        (datum, rowIndex) => mapRow(
-            (column, columnIndex) => {
-                const active = isActiveCell(activeCell, rowIndex + offset.rows, columnIndex + offset.columns);
+    ): JSX.Element[][] => {
+        const localizedFormatter = getFormatter(defaultLocale);
+        const formatters = R.map(localizedFormatter, columns);
```

Comment:
Get the formatters for all the columns from the start -- d3's formatting functions generation is not expensive but over thousands of cells it starts to have an impact.